DOL.WHD.Section14c.Api/Controllers/AttachmentController.cs

```diff
@@ -49,11 +50,18 @@ public AttachmentController(IAttachmentService attachmentService, IIdentityServi
             }
 
             var filesReadToProvider = await Request.Content.ReadAsMultipartAsync(new RestrictedMultipartMemoryStreamProvider());
+
             var files = new List<Domain.Models.Submission.Attachment>();
+            var allowedMaximumContentLength = AppSettings.Get<int>("AllowedMaximumContentLength");
             foreach (var stream in filesReadToProvider.Contents)
             {
-                if (stream.Headers.ContentLength == 0)
-                    return BadRequest("Invalid file.");
+                // The code that handle the max allowed length at the IIS level within web.config.
+                // as well as within the httpRuntime attribute found in the system.web section
+                if (stream.Headers.ContentLength < 1 || stream.Headers.ContentLength > allowedMaximumContentLength) 
+                {
+                    // File exceeds the file maximum size or empty
+                    return BadRequest("Invalid file size.");
+                }
```

Comment:
I don't see any new tests around this file size functionality, but now that I look again, I'm not seeing any tests for the `DOL.WHD.Section14c.Api` namespace at all.  🤔   Well, that's for a different issue, then.

Reply:
@mgwalker appreciate your eyes.  Good point on content length. @dciceman  We want to check buffer's byte length wherever we are reading from the incoming stream and creating the file.  (2) future multi file uploads  - i am assuming the same - @diceman for now documenting Greg's observation as a comment. (3) tests for APIs - good call on unit tests - we'll have to figure this one out.

Reply:
The current design is only allow user to upload one attachment at a time. Noted on the buffer's byte length.

Reply:
@mmurthydol Yep, exactly what I was thinking about counting bytes as they're written to disk.  Otherwise they have to be read into memory, which is probably okay since we're talking about a few megabytes, but it'd be pretty great if we didn't have to do that either - straight from the buffer to disk!  Unblock that I/O!